fix example codes (2018-04)

maiha opened the pull request:
I ran and updated example codes. (It maybe the first time in a year 😸 )
- checked by https://github.com/maiha/crystal-examples

#### master: 0.24.2+211 [ec423eb4a]
```
Specs 148 (112 successes, 15 failures, 21 pending)
Examples 1332 (856 successes, 255 failures, 221 pending)
```

#### PR
```
Specs 148 (127 successes, 0 failures, 21 pending)
Examples 1331 (1106 successes, 0 failures, 225 pending)
```

### NOTES

I will explain the points I was worried about below.

#### regex/match_data.cr 

- https://github.com/crystal-lang/crystal/blob/master/src/regex/match_data.cr#L165
```crystal
"Crystal".match(/(?<ok>Cr)|(?<ok>al)/).not_nil!["ok"]? # => "al"
```

This order would be non-deterministic, since I got `"Cr"` in my environment. It probably depends on the implementation of `LibPCRE`.

#### weak_ref.cr

- https://github.com/crystal-lang/crystal/blob/master/src/weak_ref.cr#L4-L7
```crystal
ref = WeakRef.new("oof".reverse)
ref.value # => "foo"
GC.collect
ref.value # => nil
````
Got "foo" at the last line in my environment.

Best regards,

maiha commented on src/http/formdata.cr:
````diff
@@ -39,7 +39,7 @@ require "./formdata/**"
 #
-# ```
+# ```shell
 # $ curl http://localhost:8085/ -F name=foo -F file=@/path/to/test.file
````
Added the **shell** style so that it will not be detected as a **crystal code**.

maiha replied:
Nice! I'll use `console` 👍

oprypin commented on src/string.cr:
````diff
@@ -3888,7 +3888,7 @@ class String
   #
   # ```
-  # "\u{1f48e} - à la carte\n".dump # => %("\\u1f48e - \\u00e0 la carte\\n")
+  # "\u{1e48e} - à la carte\n".dump # => %("\\u{1E48E} - \\u00E0 la carte\\n")
````
changing 1f48e to 1e48e is invalid

oprypin replied:
It's actually the opposite.
1E48E is a meaningless codepoint and 1F48E is a gem
http://www.fileformat.info/info/unicode/char/1F48E/index.htm

maiha replied:
Oh, I'm silly. :cold_sweat:
Should we use `1F48E` both in here and above example, right?

oprypin replied:
Haven't looked at it but probably

maiha commented:
pushed -f with
- fixed to use `1F48E` rather than `1E48E` for the gem marks (thanks to @oprypin )
- fixed to use `console` rather than `shell` for the command syntax in code blocks (thanks to @MakeNowJust )

MakeNowJust commented:
> **regex/match_data.cr**
>
> - https://github.com/crystal-lang/crystal/blob/master/src/regex/match_data.cr#L165
>
> ```crystal
> "Crystal".match(/(?<ok>Cr)|(?<ok>al)/).not_nil!["ok"]? # => "al"
> ```
>
> This order would be non-deterministic, since I got `"Cr"` in my environment. It probably depends on the implementation of `LibPCRE`.

I think this result is `"Cr"` always and it is a documentation bug. I'll fix this later. Thank you.

jhass commented on src/http/multipart/builder.cr:
````diff
@@ -25,6 +25,7 @@ module HTTP::Multipart
     # ```
+    # io = IO::Memory.new
     # builder = HTTP::Multipart::Builder.new(io, "a4VF")
     # builder.content_type("mixed") # => "multipart/mixed; boundary=\"a4VF\""
````
Actually not sure about this one, I think the example is meant to show to pass any io and an empty one would actually be the extraordinary case.

maiha replied:
Basically I like the minimally closed executable code to use the function. Therefore, although I added code fragments as above, I agree that it is important not to misunderstand the essence. Should we remove it? or keep it with adding some comments?

jhass commented on src/string.cr:
```diff
@@ -1585,7 +1585,7 @@ class String
   # "aabbcc".tr("abc", "xyz") # => "xxyyzz"
   # "aabbcc".tr("abc", "x")   # => "xxxxxx"
-  # "aabbcc".tr("a", "xyz")   # => "xxbbcc"
+  # "aabbcc".tr("a", "xyz")   # => "xyzxyzbbcc"
```
Don't want to derail this... but isn't that actually a bug in the `tr` implementation? Or did we intentionally diverge from Ruby's behaviour here?

maiha replied:
I don't know what specification is right. I'll create an issue about this. Thanks! 😄

maiha replied:
Oh, MakeNowJust fixed quickly! I will revert this. 😸

maiha commented:
pushed -f again with

- reverted `String#tr` (related: #5913)
- added comments about using `IO::Memory` in `HTTP::Multipart::Builder`

oprypin commented:
You wouldn't need to describe what you're pushing, if you just ... pushed it.
And the reviewers wouldn't need to review everything again.
There is no advantage to squashing the commits yourself because they can be squashed by GitHub.
